### PR TITLE
chore: change from `go-playground/validator` to Manual Validation for settings

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Removing `go-playground/validator` in favor of manual validation 

**How was this change tested?**
-  `/karpenter snapshot`
 
**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: pkg/apis/settings/settings_validation.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		s.validateTags(),
+		s.validateClusterName(),
+		s.validateVMMemoryOverheadPercent(),
+		s.validateReservedENIs(),
+		s.validateAssumeRoleDuration(),
+	)
+}
+
+func (s Settings) validateAssumeRoleDuration() error {
+	if s.AssumeRoleDuration == 0 {
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:40]: I think just checking the minimum here should probably be enough. This won't be 0 unless someone explicitly sets it to 0 due to our defaults.

**Review by jonathan-innis — COMMENTED** [review]
Just one comment on testing a value for 0

File: pkg/apis/settings/settings_validation.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		s.validateClusterName(),
+		s.validateVMMemoryOverheadPercent(),
+		s.validateReservedENIs(),
+		s.validateAssumeRoleDuration(),
+	)
+}
+
+func (s Settings) validateAssumeRoleDuration() error {
+	if s.AssumeRoleDuration < time.Minute*15 {
+		return fmt.Errorf("AssumeRoleDuration cannot be less then 15 Minutes")
```
Comment (njtran) [inline pkg/apis/settings/settings_validation.go:41]: nit:
```suggestion
		return fmt.Errorf("AssumeRoleDuration cannot be less than 15 Minutes")
```

File: pkg/apis/settings/settings_validation.go
Comment (njtran) [inline pkg/apis/settings/settings_validation.go:1]: consider using apis.FieldError() here like we do in Karpenter-core: https://github.com/aws/karpenter-core/blob/main/pkg/apis/v1beta1/nodeclaim_validation.go#L61

File: pkg/apis/settings/settings_validation.go [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+func (s Settings) Validate() (errs *apis.FieldError) {
+	return errs.Also(
+		s.validateEndpoint(),
+		s.validateTags(),
+		s.validateClusterName(),
+		s.validateVMMemoryOverheadPercent(),
+		s.validateReservedENIs(),
+		s.validateAssumeRoleDuration(),
+	)
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:35]:
```suggestion
	).Via("aws")
```

This allows you to scope all of the field names that you are using in the checks in this set

File: pkg/apis/settings/settings_validation.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+func (s Settings) validateAssumeRoleDuration() (errs *apis.FieldError) {
+	if s.AssumeRoleDuration < time.Minute*15 {
+		return errs.Also(apis.ErrInvalidValue("assumeRoleDuration cannot be less than 15 Minutes", "assumeRoleDuration"))
+	}
+	return nil
+}
+
+func (s Settings) validateClusterName() (errs *apis.FieldError) {
+	if s.ClusterName == "" {
+		return errs.Also(apis.ErrInvalidValue("clusterName is required", "clusterName"))
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:47]: I think there _might_ be a required values error type in the `apis` package

File: pkg/apis/settings/settings_validation.go [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+func (s Settings) validateEndpoint() (errs *apis.FieldError) {
+	if s.ClusterEndpoint == "" {
+		return nil
+	}
+	endpoint, err := url.Parse(s.ClusterEndpoint)
+	// url.Parse() will accept a lot of input without error; make
+	// sure it's a real URL
+	if err != nil || !endpoint.IsAbs() || endpoint.Hostname() == "" {
+		return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("\"%s\" not a valid clusterEndpoint URL", s.ClusterEndpoint), "clusterEndpoint"))
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:60]:
```suggestion
		return errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%q not a valid clusterEndpoint URL", s.ClusterEndpoint), "clusterEndpoint"))
```

File: pkg/apis/settings/settings_validation.go [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+	}
+	return nil
+}
+
+func (s Settings) validateTags() (errs *apis.FieldError) {
+	for k := range s.Tags {
+		for _, pattern := range v1alpha1.RestrictedTagPatterns {
+			if pattern.MatchString(k) {
+				errs = errs.Also(errs, apis.ErrInvalidKeyName(k, "tags", fmt.Sprintf("tag contains a restricted tag %q", pattern.String())))
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:69]:
```suggestion
				errs = errs.Also(errs, apis.ErrInvalidKeyName(k, "tags", fmt.Sprintf("tag contains a restricted tag matching the pattern %q", pattern.String())))
```

File: pkg/apis/settings/settings_validation.go [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+				errs = errs.Also(errs, apis.ErrInvalidKeyName(k, "tags", fmt.Sprintf("tag contains a restricted tag matching the pattern %q", pattern.String())))
+			}
+		}
+	}
+	return errs
+}
+
+func (s Settings) validateVMMemoryOverheadPercent() (errs *apis.FieldError) {
+	if s.VMMemoryOverheadPercent < 0 {
+		return errs.Also(apis.ErrInvalidValue("cannot be negative", "VMMemoryOverheadPercent"))
```
Comment (jonathan-innis) [inline pkg/apis/settings/settings_validation.go:78]:
```suggestion
		return errs.Also(apis.ErrInvalidValue("cannot be negative", "vmMemoryOverheadPercent"))
```

**Review by jonathan-innis — APPROVED** [review]
LGTM 🚀

**Pull request merged**